### ISSUE TITLE
Allow async try/catch on ECONNREFUSED

### ISSUE THREAD

**Issue by rightisleft** [typeorm/typeorm#198]
Wrapping a try catch around createConnection() does not properly reject low level errors.

Step 1: Kill db daemon to ensure its not running 
Step 2: call the following:

```
    try {
      if (!instance) {
        this._connectionOptions = {
          name: this._databaseName,
          driver: this._driverOptions,
          entities: [Foo,Bar,Baz],
          autoSchemaSync: true
        };
        instance = await createConnection(this._connectionOptions);
      }
    } catch (e) {
     console.error("Cannont Connect ");
    }

```
Step 3: Note that because the error is not caught - it bubbles up and does not hit the catch block.

```
Error: connect ECONNREFUSED 127.0.0.1:5432
    at Object.exports._errnoException (util.js:1026:11)
    at exports._exceptionWithHostPort (util.js:1049:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1085:14)
[nodemon] app crashed - waiting for file changes before starting...
```

**Comment by pleerock** [comment]
Cannot reproduce. There is example Ive run:

```typescript
import "reflect-metadata";
import {createConnection, ConnectionOptions} from "typeorm";
import {Post} from "./entity/Post";

const options: ConnectionOptions = {
    driver: {
        type: "postgres",
        host: "localhost",
        port: 54321,
        username: "root",
        password: "admin",
        database: "test"
    },
    logging: {
        logQueries: true,
        logSchemaCreation: true
    },
    autoSchemaSync: true,
    entities: [Post]
};

async function a() {
    try {
        await createConnection(options);
    } catch (err) {
        console.log("Caught! Cannot connect: ", err);
    }
}

a();
```

And result:

```
$ node app.js
Caught! Cannot connect:  { Error: connect ECONNREFUSED 127.0.0.1:54321
    at Object.exports._errnoException (util.js:1036:11)
    at exports._exceptionWithHostPort (util.js:1059:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1080:14)
  code: 'ECONNREFUSED',
  errno: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 54321 }
```

**Comment by joffray589** [comment]
I had the same problem and found a quick fix, here is the code.
`(async () => {
    try{
        let connection = await createConnection(dbOptions);
        await connection.getRepository(User).find(); // this will throw ECONNREFUSED 
        console.log("Connected to mysql");
        await new BackendWebServer(connection).listen(config.server.listenPort);
        console.log("BackendWebServer listening on port " + config.server.listenPort);
    }
    catch(error){
        console.error("Error on server start : " + error );
    }
})();`

And the output : 
`Error on server start : Error: connect ECONNREFUSED 127.0.0.1:3306`

**Comment by joffray589** [comment]
I had the same problem and found a quick fix, here is the code.
```
(async () => {
    try{
        let connection = await createConnection(dbOptions);
        await connection.getRepository(User).find(); // this will throw ECONNREFUSED 
        console.log("Connected to mysql");
        await new BackendWebServer(connection).listen(config.server.listenPort);
        console.log("BackendWebServer listening on port " + config.server.listenPort);
    }
    catch(error){
        console.error("Error on server start : " + error );
    }
})();
```
And the output : 
```
Error on server start : Error: connect ECONNREFUSED 127.0.0.1:3306
```

**Comment by pleerock** [comment]
@joffray589 can you create a sample repo, so I can reproduce this issue and implement  a proper fix for it?

**Comment by joffray589** [comment]
@pleerock I created a sample repo (https://github.com/joffray589/typeorm-sample) but I can't reproduce the issue on it. 
I'm gonna investigate to find what is the difference beetween this sample and my main project on next week.
I can also give you an access to my main project and write a little doc to explain how to reproduce the issue on it, just tell me if you want to.
(sorry if my english is bad)

**Comment by pleerock** [comment]
better if you can provide issue on a sample repo

**Comment by pleerock** [comment]
Please create a new issue once you could reproduce your issue. I'll close this one.
